realm/src/androidTest/java/io/realm/internal/JNIQueryTest.java

```diff
@@ -315,19 +315,19 @@ public void testNegativeColumnIndexContains() {
         TableQuery query = table.where();
 
         // String
-        try { query.contains(-1, "hey").findAll(); fail("-1 column index"); } catch (ArrayIndexOutOfBoundsException e) {}
-        try { query.contains(-1, "hey").findAll(); fail("-10 column index"); } catch (ArrayIndexOutOfBoundsException e) {}
-        try { query.contains(-1, "hey").findAll(); fail("-100 column index"); } catch (ArrayIndexOutOfBoundsException e) {}
+        try { query.contains(new long[]{-1}, "hey").findAll(); fail("-1 column index"); } catch (ArrayIndexOutOfBoundsException e) {}
```

Comment:
You are absolutely right - the unit tests for the JNI functions could be much better and they haven't been touch for a very long time.